File: Backend/src/main/java/com/gameshub/model/request/ProductRequestDAO.java
package com.gameshub.model.request;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDate;

@Getter
@Setter
@NoArgsConstructor
@MappedSuperclass
public abstract class ProductRequestDAO {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "Id")
    protected int id;

    @Column(name = "Datereceived")
    protected LocalDate dateReceived;

    @Column(name = "Status")
    protected String status;

    @Column(name = "Requesttype")
    protected String requestType;

    @Column(name = "Title")
    protected String title;

    @Column(name = "Price")
    protected float price;

    @Column(name = "Description")
    protected String description;

    @Column(name = "Postdate")
    protected LocalDate postDate;

    @Column(name = "Count")
    protected int count;

    @Column(name = "Category")
    protected String category;

    @Column(name = "Sellerid")
    protected int sellerId;

}

File: Frontend/src/Models/product_request/ProductRequestDAO.ts
import { ProductDAO } from "../product/ProductDAO"

export class ProductRequestDAO extends ProductDAO{
    constructor(
        id: number,
        seller: any,
        title: string,
        // imageURLs: string[],
        price: number,
        description: string,
        postDate: number[],
        count: number,
        productType: string,
        category: string,
        public status: string,
        public requestType: string
        ){
            super(id, seller, title, price, description, postDate, count, productType, category)
    }

}